app/controllers/search_controller.rb

```diff
@@ -55,6 +55,7 @@ def index
       else
         if params[:body_is_like] == '1' && !user_signed_in?
           flash[:warning] = 'Unregistered users cannot use LIKE searches on the body field. Please sign in.'
+          redirect_to(root_path) && return
```

Comment:
You don't need a `return` statement, `redirect_to` automatically terminates the request.